packages/babel-plugin-proposal-destructuring-private/test/fixtures/assignment--es2015/array-rest/output.js

```diff
@@ -0,0 +1,14 @@
+let x, z;
+
+class C {}
+
+var _x = {
+  writable: true,
+  value: void 0
+};
+
+(() => {
+  var _m, _p, _p2, _m2;
+
+  _m = [C], [_p, ..._p2] = _m, _m2 = babelHelpers.classStaticPrivateFieldSpecGet(_p, C, _x), x = _m2 === void 0 ? 1 : _m2, z = _p2;
```

Comment:
`_m` is not needed here, maybe we could split the assignment only if we need to reference it multiple times.